tests/common/include/tests/common/read_geometry.hpp

```diff
@@ -170,6 +177,69 @@ struct object_registry {
     }
 };
 
+// Minimalistic geometry type for toy geometry.
+template <typename volume_t, typename object_t,
+          template <typename...> class vector_type = dvector>
+struct toy_geometry {
+    // typedefs
+    using volume_type = volume_t;
+    using portal = object_t;
+    using portal_container = dvector<portal>;
+    using portal_links = typename object_t::edge_links;
+
+    struct object_registry {
+        using id = typename volume_type::objects;
+
+        template <typename value_type = void>
+        static constexpr auto get() {
+            return id::e_surface;
+        }
+    };
+
+    toy_geometry(vector_type<volume_t>&& volumes,
+                 vector_type<object_t>&& objects)
+        : _volumes(volumes), _objects(objects) {}
+
+    // data containers
+    vector_type<volume_t> _volumes;
+    vector_type<object_t> _objects;
+};
+
+// Minimalistic detector type for toy geometry
+template <typename geometry_t, typename transform_container_t,
+          typename mask_container_t>
+struct toy_detector {
```

Comment:
Hmm so you decided to write a stand alone detector class for toy_detector - it's OK. But later we may have to pay for this later.

```c++
template < typename detector_t >
struct navigator {
   using detector_type = detector_t;
   navigator(detector_t d): det(d){}
   detector_t det; 
} ;

template < typename navigator_t >
struct navigator_data{
  navigator_data ( navigator_t nav ): det_data(nav.det){}
  detector_data< typename navigator_t::detector_type > det_data; 
};
```
To make above code work, we will have to define `detector_data` separately for `toy_detector` and `detector`

Reply:
I just checked out the member variables of `detector`, which seems to have same naming - so I think we can apply this concept. Can you just add a dummy `volume_grid` member?

Reply:
Most of that existed in the previous PR already. I had hoped that we might use the same memory management later, as this mainly leaves out what is not needed on device anyway (I hope :))

Reply:
Added ```volume_grid``` and ```surface_finder``` as a placeholder, until I get around to build some